R/download_data_factors.R

```diff
@@ -71,16 +74,21 @@ download_data_factors <- function(type, start_date, end_date) {
 #' @importFrom lubridate ymd floor_date
 #'
 #' @export
-download_data_factors_ff <- function(type, start_date, end_date) {
+download_data_factors_ff <- function(
+    type, start_date = NULL, end_date = NULL
+  ) {
 
   check_supported_type(type)
 
   check_if_package_installed("frenchdata", type)
-
   download_french_data <- getNamespace("frenchdata")$download_french_data
 
-  start_date <- as.Date(start_date)
-  end_date <- as.Date(end_date)
+  if (is.null(start_date) || is.null(end_date)) {
+    message("No start_date or end_date provided. Returning the full data set.")
+  } else {
+    start_date <- as.Date(start_date)
+    end_date <- as.Date(end_date)
+  }
```

Comment:
Why not merge this part with the `filter(between(...))`-part? I suggest moving it down.

Reply:
Because the user sees the message before the data download starts and may kill the job before everything is prepared and filtered. This is what I'd want to have, so I'll kindly reject :)